packages/form-footer/src/FormFooter.tsx

```diff
@@ -119,60 +114,41 @@ export default function FormFooter({
   contentClassName,
   className,
 }: FormFooterProps) {
-  const RenderedPrimaryButton: React.ReactNode = useMemo(() => {
-    if (primaryButton) {
-      if (isPrimaryButtonProps(primaryButton)) {
-        return (
+  return (
+    <footer className={cx(footerStyle, className)}>
+      <div className={cx(contentStyle, contentClassName)}>
+        {backButtonText && (
           <Button
-            variant={primaryButton.variant ?? 'primary'}
-            disabled={primaryButton.disabled}
-            onClick={primaryButton.onClick}
-            type={primaryButton.type}
+            variant="default"
+            onClick={onBackClick}
             className={buttonStyle}
+            leftGlyph={<ChevronLeftIcon />}
           >
-            {primaryButton.text}
+            {backButtonText}
           </Button>
-        );
-      }
-
-      if (isComponentType(primaryButton, 'Button')) {
-        return primaryButton;
-      }
-
-      errorOnce(
-        '`primaryButton` prop in `FormFooter` must be either a `Button` component, or object with at minumum a `text` property',
-      );
-    }
-  }, [primaryButton]);
-
-  return (
-    <footer className={cx(footerStyle, className)}>
-      <div className={cx(contentStyle, contentClassName)}>
-        <span className={flexSpan}>
-          {backButtonText && (
+        )}
+        <div className={flexEndContent}>
+          {errorMessage && (
+            <Banner className={bannerStyle} variant="danger">
+              {errorMessage}
+            </Banner>
+          )}
+          {cancelButtonText && (
             <Button
               variant="default"
-              onClick={onBackClick}
+              onClick={onCancel}
               className={buttonStyle}
             >
-              {backButtonText}
+              {cancelButtonText || 'Cancel'}
             </Button>
           )}
-          {errorMessage && (
-            <Banner className={bannerStyle} variant="danger">
-              {errorMessage}
-            </Banner>
+          {isComponentType(primaryButton, 'Button') ? (
+            primaryButton
+          ) : (
+            <PrimaryButton {...(primaryButton as PrimaryButtonProps)} />
```

Comment:
What happens if a random HTML element is passed, say like a `span`?

Reply:
Ah that's a good catch. I've brought back the code to raise an error once to the `PrimaryButton` component!